content/cucumber/step-definitions.md

```diff
@@ -253,3 +253,48 @@ Given("I have {int} {color} balls", function (int1, color) {
 });
 ```
 {{% /block %}}
+
+# Abstract Your Steps
+
+It is always a good idea to make sure that you are not writing duplicate/similar step definitions. While there are a number of ways to do so and you can document your step definitions as well, it is always good to make use of **helper methods** to abstract your step definitions and reduce the number of steps.
+
+For example, if you have the following steps used across multiple scenarios :
+
+```
+    Given I go to the home page
+    Given I check the about page of the website 
+    Given I get the contact details
+```
+
+and all of these steps have different step definitions which visit the Home, About and Contact pages, you might be writing _redundant steps_. It is possible that the underlying code for executing each of these test cases is different, **but** it is important to understand that from the point of view of their **behavior**, all of these steps do essentially similar things, ie.
+
+_* Open the Home page_
+_* Open the About page_
+_* Open the Contact page_
+
+As such, you can use abstract helper methods to reduce all these into one step :
+
+    Given I go to the {} page
+
+with the step def :
+
+```java
```

Comment:
We'll need to add examples for the other languages too. I can do Kotlin, we ask help for javascript & ruby if needed.

Reply:
Yes that would be great :)

I am only familiar with Java so I didn't write one for the other two.

Reply:
That's fine. We'll get there.
You could add code blocks for javascript, ruby, kotlin with a `// TODO` inside and we'll get to them later.

Reply:
Done